lib/spack/spack/installer.py

```diff
@@ -1702,6 +1702,16 @@ def install(self):
                 spack.hooks.on_install_cancel(task.request.pkg.spec)
                 raise
 
+            except binary_distribution.NoChecksumException as exc:
+                if not task.request.install_args.get("cache_only"):
```

Comment:
```suggestion
                if not task.cache_only:
```

ref. #32537

Reply:
I also noticed another place that was missed in the BuildRequest class, ln:2360

```python
         deptypes = ["link", "run"]
         include_build_deps = self.install_args.get("include_build_deps")
         if not self.install_args.get("cache_only") or include_build_deps:
             deptypes.append("build")
         if self.run_tests(pkg):
             deptypes.append("test")
         return tuple(sorted(deptypes))
```

It should probably check `dependencies_cache_only` there instead.